lib/testutils.g

```diff
@@ -0,0 +1,264 @@
+######################################################################
+#
+# Checking the primitive groups of degree n
+#
+######################################################################
+#
+# 1. Consistency check: each group of degree n should act
+# transitively and primitively on [1..n]
+#
+
+######################################################################
+#
+# Consistency check for a given degree
+#
+PrimGrpConsistencyCheckDegree := function(deg) 
+    local i, g;
+    for i in [1..NrPrimitiveGroups(deg)] do
+        g := PrimitiveGroup(deg,i);
+        if MovedPoints(g) <> [1..deg] then
+            Print("PrimitiveGroup(",deg,",",i,") is not acting on ", [1..deg], "\n");
+        fi;
+        if not IsTransitive(g,[1..deg]) then
+            Print("PrimitiveGroup(",deg,",",i,") is not transitive on ", [1..deg], "\n");
+        fi;
+        if not IsPrimitive(g,[1..deg]) then
+            Print("PrimitiveGroup(",deg,",",i,") is not primitive on ", [1..deg], "\n");
+        fi;
+    od;
+end;
+
+######################################################################
+#
+# Consistency check for a range of degrees
+#
+PrimGrpConsistencyCheck := function(i,j)
+local deg;
+for deg in [i..j] do
+    Print("Degree ", deg, " : ", NrPrimitiveGroups(deg), " groups \r");
+    PrimGrpConsistencyCheckDegree(deg);
+od;
+Print("                                                           \r");
+end;
+
+######################################################################
+#
+# 2. Checking names of primitive groups
+#
+
+######################################################################
+#
+# Constructors for some groups
+#
+AGL := {d,q} -> SemidirectProduct(GL(d, GF(q)), GF(q)^d);
+ASL := {d,q} -> SemidirectProduct(SL(d, GF(q)), GF(q)^d);
+L   := {d,q} -> SimpleGroup("L", d, q);
+
+ASigmaL := function(d, q)
+    local G, V;
+    G := SigmaL(d, q);
+    V := FieldOfMatrixGroup(G) ^ DimensionOfMatrixGroup(G);
+    return SemidirectProduct(G, V);
+end;
+
+AGammaL := function(d, q)
+    local G, V;
+    G := GammaL(d, q);
+    V := FieldOfMatrixGroup(G) ^ DimensionOfMatrixGroup(G);
+    return SemidirectProduct(G, V);
+end;
+
+PSigmaL := function(d, q)
+    local G, gens;
+    G := SigmaL(d, q);
```

Comment:
This function and PGammaL are wrong for `d=1`, in that case the trivial group should be returned.

Reply:
done